domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt

```diff
@@ -227,53 +226,31 @@ class TopicController @Inject constructor(
   }
 
   private fun checkIfTopicIsOngoing(topic: Topic, topicProgress: TopicProgress): Boolean {
-    val completedChapterProgressList = ArrayList<ChapterProgress>()
-    val startedChapterProgressList = ArrayList<ChapterProgress>()
-    topicProgress.storyProgressMap.values.toList().forEach { storyProgress ->
-      completedChapterProgressList.addAll(
-        storyProgress.chapterProgressMap.values
-          .filter { chapterProgress ->
-            chapterProgress.chapterPlayState ==
-              ChapterPlayState.COMPLETED
-          }
-      )
-      startedChapterProgressList.addAll(
-        storyProgress.chapterProgressMap.values
-          .filter { chapterProgress ->
-            chapterProgress.chapterPlayState ==
-              ChapterPlayState.STARTED_NOT_COMPLETED
-          }
-      )
-    }
-
-    // If there is at least 1 completed chapter and 1 not-completed chapter, it is definitely an
-    // ongoing-topic.
-    if (startedChapterProgressList.isNotEmpty()) {
-      return true
+    // If there's at least one story with progress and not yet completed, then the topic
+    // is considered ongoing.
+    return topic.storyList.any { storySummary ->
+      topicProgress.storyProgressMap[storySummary.storyId]?.let { storyProgress ->
+        storySummary.isOngoing(storyProgress)
+      } ?: false
     }
+  }
 
-    topic.storyList.forEach { storySummary ->
-      if (topicProgress.storyProgressMap.containsKey(storySummary.storyId)) {
-        val storyProgress = topicProgress.storyProgressMap[storySummary.storyId]
-        val lastChapterSummary = storySummary.chapterList.last()
-        if (!storyProgress!!.chapterProgressMap.containsKey(lastChapterSummary.explorationId)) {
-          return true
-        }
-      }
-    }
+  /** Return whether the current [StorySummary] can be considered "ongoing" given the specified
+   * [StoryProgress] (that is, at least one chapter has started and the final chapter isn't yet
+   * completed).
+   */
```

Comment:
```suggestion
  /**
   * Returns whether the current [StorySummary] can be considered "ongoing" given the specified
   * [StoryProgress] (that is, at least one chapter has started and the final chapter isn't yet
   * completed).
   */
```

Note the formatting here for consistency. Please update the one below, too.

Reply:
Done